### ISSUE TITLE
Sentintel5 files are incorrectly downloaded as zip files when they are actually nc files

### ISSUE THREAD

**Issue by craigmillernz** [sentinelsat/sentinelsat#270]
When I download Sentinel5 files they are downloaded as *.zip.  However they are not zip files, but are netcdf files and should have the extension *.nc.

**Comment by j08lue** [comment]
Yes, this is a bug - we hardcode the extension here:

https://github.com/sentinelsat/sentinelsat/blob/f90486366cb82b51379251707d0ca81a7c2e2a4c/sentinelsat/sentinel.py#L494

**Comment by j08lue** [comment]
Do you want to try to fix it, @craigmillernz? I guess the correct extension is available from the `product_info` somewhere...

**Comment by valgur** [comment]
Kind of surprisingly, the real filename is not included in either Solr or OData API responses. The 'filename' attribute for most products has a '.SAFE' extension rather than the true '.zip'.

There's always the `Content-Disposition` header, though. We could do something like:
```python
from cgi import parse_header

response = session.head(url)
filename = parse_header(response.headers.get('content-disposition'))[-1].get('filename')
```

Might be a good time to add path sanitizing to the received filename as well as a good practice, in case the server happens to return something malicious. This library looks useful: https://pypi.org/project/pathvalidate/

**Comment by craigmillernz** [comment]
The filename is listed in the product dictionary, complete with extension, could just be a simple matter of grabbing that?  Adding some filename validation could be useful as well.

`'filename': 'S5P_NRTI_L2__SO2____20190328T020752_20190328T021252_07529_01_010105_20190328T025915.nc'`

**Comment by valgur** [comment]
The OData info does not provide the correct downloaded product's filename for most cases besides Sentinel 5, unfortunately. It states '*.SAFE' for S1 and S2 and '*.SEN3' for S3 as the filenames even though the downloadable files are .zip-s for all of them.

**Comment by j08lue** [comment]
@valgur's suggestion with the `Content-Disposition` header seems like the way to go.

**Comment by lkiesow** [comment]
I can take a look at this…

**Comment by ahadmushir** [comment]
Hello, I would like to work on this bug fix for Hacktoberfest.

**Comment by Fernerkundung** [comment]
@ahadmushir that sounds great :+1

**Comment by ahadmushir** [comment]
> @ahadmushir that sounds great 👍
> 
> Do you need more info to get started on this or is this issue sufficient?

A little more info would be nice, Sorry was out of touch (caught in the weather).

**Comment by valgur** [comment]
I think this issue has been already resolved in #320 and #328 and probably does not need further work. I just have not gotten around to wrapping up and merging the PR, sorry about that.
